lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -180,10 +180,9 @@ def _put(self, params: ApiParams, _) -> Response:
             try:
                 Metadata.setvalue(key=k, value=v, dataset=dataset, user_id=user_id)
             except MetadataError as e:
-                self.logger.warning("Unable to update key {} = {!r}: {}", k, v, str(e))
-                fail = True
+                fail.append(str(e))
 
         if fail:
-            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
+            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR, ", ".join(fail))
```

Comment:
Prior to this change, all the warning messages we only logged, and now they are all being returned in the "INTERNAL ERROR" message of the API call.  This seems like a bad idea.

Reply:
No, they were previously logged but *also* failed with an internal error. Now the internal error has a message.

Reply:
Exactly, that is what I am objecting to.

Reply:
What you wrote suggested otherwise. OK, so you just don't like telling the client anything. Even for a "real" internal error that's pretty unhelpful, but there's often nothing specific we can say.

This is actually an odd "partial success" situation, where I think it's necessary, and in HTTP-speak there's really no obviously better way to represent this than a 500. I'm kinda starting to think there's no way to move forward on this until we can ditch sqlite3 and I can work in a nested transaction here to avoid the partial success.

But maybe it's time to take up some mechanism to tie the log and internal error messages together... which would help in the usual cases, though not so much here. 😦